src/legacy/core_plugins/kibana/public/dashboard/application.ts

```diff
@@ -0,0 +1,228 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiConfirmModal, EuiIcon } from '@elastic/eui';
+import angular, { IModule } from 'angular';
+import { IPrivate } from 'ui/private';
+import { i18nDirective, i18nFilter, I18nProvider } from '@kbn/i18n/angular';
+import {
+  AppMountContext,
+  ChromeStart,
+  LegacyCoreStart,
+  SavedObjectsClientContract,
+  UiSettingsClientContract,
+} from 'kibana/public';
+import { Storage } from '../../../../../plugins/kibana_utils/public';
+import {
+  GlobalStateProvider,
+  StateManagementConfigProvider,
+  AppStateProvider,
+  PrivateProvider,
+  EventsProvider,
+  PersistedState,
+  createTopNavDirective,
+  createTopNavHelper,
+  PromiseServiceCreator,
+  KbnUrlProvider,
+  RedirectWhenMissingProvider,
+  confirmModalFactory,
+  configureAppAngularModule,
+} from './legacy_imports';
+
+// @ts-ignore
+import { initDashboardApp } from './legacy_app';
+import { DataStart } from '../../../data/public';
+import { SavedQueryService } from '../../../data/public/search/search_bar/lib/saved_query_service';
```

Comment:
you shouldn't be importing from deep inside the plugins (only top level)

Reply:
Good catch, fixed